### PR TITLE
Transactional queue contains all operation

### PR DESCRIPTION
**Pull request by alex-dukhno**
Should be merged after #16129

### CODE REVIEW

**Comment by mmedenjak** [comment]
run-lab-run

**Comment by ihsandemir** [comment]
what is the protocol PR link?

File: hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java [hunk 1]
```diff
@@ -372,6 +392,46 @@ public boolean txnCommitOffer(long itemId, Data data, boolean backup) {
...
+                getBackupMap().put(itemId, item);
+            }
+            i++;
+            toStore.put(itemId, datum);
+        }
+        if (store.isEnabled() && !backup) {
+            try {
+                store.storeAll(toStore);
+            } catch (Exception e) {
+                logger.warning("Exception during store", e);
```
Comment (ihsandemir) [inline hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java]: should this return failure `false`?

File: hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java [hunk 2]
```diff
@@ -372,6 +392,46 @@ public boolean txnCommitOffer(long itemId, Data data, boolean backup) {
...
+            }
+        }
+        return true;
+    }
+
+    public boolean txnCommitContainsAll(List<Data> data) {
+        for (Data datum : data) {
+            if (getItemQueue().stream().noneMatch(item -> datum.equals(item.getData()))) {
```
Comment (ihsandemir) [inline hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java]: why not cache `getItemQueue().stream()` ?

File: hazelcast/src/test/java/com/hazelcast/collection/impl/txnqueue/TransactionQueueBulkOperationsTest.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        queue = context.getQueue(queueName);
+        queue.addAll(1, 2, 3);
+
+        context.commitTransaction();
+
+        context = instance.newTransactionContext();
+        context.beginTransaction();
+        try {
+            queue = context.getQueue(queueName);
+            assertTrue(queue.containsAll(asList(1, 2)));
```
Comment (ihsandemir) [inline hazelcast/src/test/java/com/hazelcast/collection/impl/txnqueue/TransactionQueueBulkOperationsTest.java]: Can we add negative test cases + cases where list is empty?

File: hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java (hunk 2 above)
Comment (pivovarit) [reply hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java]: Streams are not reusable

Comment (ihsandemir) [reply hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java]: I would think that performance is important.

**Comment by mmedenjak** [comment]
@alex-dukhno in addition to the comments, please rebase and address the checkstyle issues:
```
08:29:08 [ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/collection/impl/txnqueue/TransactionalQueueProxySupport.java:44:8: Unused import - java.util.ArrayList. [UnusedImports]
08:29:08 [ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/collection/impl/txnqueue/TransactionalQueueProxySupport.java:45:8: Unused import - java.util.Arrays. [UnusedImports]
08:29:08 [ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/collection/impl/txnqueue/TransactionalQueueProxySupport.java:51:8: Unused import - java.util.stream.Collectors. [UnusedImports]
```

**Comment by alex-dukhno** [comment]
> Please clean addAll changes from this PR.

@mmedenjak @ihsandemir I will rebase over master when I have a time, maybe the PR with addAll methods will be merged before

File: hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java (hunk 1 above)
Comment (alex-dukhno) [reply hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java]: Code is used by `TransactionalQueue.addAll` will address in #16129

**Comment by mmedenjak** [comment]
Moving into 4.1 as we have reached the deadline for 4.0. Don't worry, we'll keep reviewing and get it merged soon. Thank you for the effort.

**Comment by pveentjer** [comment]
Closing this PR due to lack of progress.

**Pull request closed without merging**